src/pages/__tests__/password-update.test.js

```diff
@@ -0,0 +1,141 @@
+import { useDataMutation } from '@dhis2/app-runtime'
+import { screen } from '@testing-library/react'
+import userEvent from '@testing-library/user-event'
+import React from 'react'
+import { useLoginConfig } from '../../providers/use-login-config.js'
+import { renderWithRouter } from '../../test-utils/render-with-router.js'
+import PasswordUpdatePage from '../password-update.js'
+
+jest.mock('../../components/not-allowed-notice.js', () => ({
+    NotAllowedNotice: () => <div>NOT ALLOWED</div>,
+}))
```

Comment:
for mocks like this, my preserence is always not to mock them .. the test becomes more like an integration rather than unit, but since most of the bugs happen when integrating components, then that's useful.  (basically the ideas from this [article](https://kentcdodds.com/blog/write-tests) I keep banging on)

This applies in other tests too.

I think though that's a larger discussion .. for now, this is great as it is. But we can bring up that and discuss it as a team.

Reply:
Thanks. Yes. These tests have a lot of mocking, and I was skewing to keeping them on the unit test side of things, which might just be because in my head I had set myself the task of writing unit tests 😊; I'm evidently impervious to taking the message from the article you've been banging on about.